ratpack-core/src/test/groovy/ratpack/http/client/HttpClientSmokeSpec.groovy

```diff
@@ -517,4 +519,73 @@ BAR
     text == "bar"
   }
 
+
+  def "can handle compressed responses"() {
+    given:
+    def stringBuf = new ByteArrayOutputStream()
+    def writer = new OutputStreamWriter(new GZIPOutputStream(stringBuf))
+    writer.write("bar")
+    writer.close()
+
+    otherApp {
+      get("foo") {
+        response.headers.add(CONTENT_ENCODING, "gzip")
```

Comment:
I couldn't find the code where this is being done, can you give me some pointers?

One thing I'm worried about is that, by not making explicit that the response is compressed, ```otherApp```'s response could be uncompressed, which would make the test pass incorrectly.